src/transformStateWithClones.js

```diff
@@ -5,9 +5,60 @@
  * @param {Object[]} actions
  *
  * @return {Object[]}
+ *
+ * transformStateWithClones()
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        result.push(addProperties(stateCopy, action.extraData));
+        break;
+
+      case 'removeProperties':
+        result.push(removeProperties(stateCopy, action.keysToRemove));
```

Comment:
Issue: The `stateCopy` object is modified directly within the `switch` cases. This can lead to unintended side effects as the same object is used for all actions.

Suggestion: Instead of pushing the result inside each case, modify the `stateCopy` within the case and push a new copy of the modified state to the `result` array after the `switch` block, at the end of each loop iteration.